Add Ecobee humidifier device_info and unique_id

bjpetit opened the pull request:
The humidifier support in the Ecobee integration did not return device_info, causing the humidifier to not be associated
with the ecobee integration or the parent thermostat.
Added unique_id property

This change also fills out the ecobee-data.json fixture data a bit to address failures in the test setup and increase coverage in the tests.

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->


## Proposed change

This change adds the device_info and unique_id properties to the ecobee humidifier support code. Without the device_info property the ecobee humidifier is not being linked in as an entity of the ecobee thermostat, and is not associated with the ecobee integration.



## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #50046

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

bjpetit committed:
Add Ecobee humidifier device_info and unique_id

Ecobee humidifier entity was not connected to the thermostat device.
This change will ensure the entitiy is properly connected.

This change also fills out the ecobee-data.json fixutre data a bit
to address failures in the test setup.

bjpetit commented:
This work was previously in #50141.

MartinHjelmare commented on tests/fixtures/ecobee/ecobee-data.json:
```diff
@@ -1,6 +1,8 @@
 {
     "thermostatList": [
-        {"name": "ecobee",
+        {"identifier": 8675309,
```
```suggestion
        {
            "identifier": 8675309,
```

bjpetit replied:
Updated

bjpetit committed:
Add Ecobee humidifier device_info and unique_id

Adjust test fixture data to increase pytest coverage
Clean up indenting in ecobee-data.json

MartinHjelmare reviewed (approved):
Looks good!

bjpetit commented:
Any recommendations on the path to take regarding the codecov issue? 
The spot it's complaining about is intended to handle unknown Ecobee models. I've tested it locally, but in order to get both cases covered, the tests would need to refactor the tests to run multiple permutations of ecobee-data.json. A good goal, but probably out of scope for this change.

bjpetit committed:
Add Ecobee humidifier device_info and unique_id

Update exception case in device_info to not
be included in codecov tests. This case has
been tested locally.

bjpetit commented:
I've added my proposed change for the codecov failure to this PR. If there is another approach that I should take, I'll be happy to adjust.

bjpetit committed:
Add Ecobee humidifier device_info and unique_id

Address pylint issue

MartinHjelmare commented on homeassistant/components/ecobee/humidifier.py:
```diff
@@ -43,6 +43,32 @@ def __init__(self, data, thermostat_index):
+        """Return device information for the ecobee humidifier."""
+        try:
+            model = f"{ECOBEE_MODEL_TO_NAME[self.thermostat['modelNumber']]} Thermostat"
+        except KeyError:  # pragma: no cover
```
It's better to exclude the whole module from coverage calculation. Please add it in `.coveragerc`.

bjpetit replied:
Ok. Updated. Thanks.

bjpetit committed:
Add Ecobee humidifier device_info and unique_id

Remove no cover pragma and add ecobee humidifier.py to
.coveragerc

MartinHjelmare reviewed (approved):
Thanks!

bjpetit commented:
Is there anything else needed from me before this PR can be merged? I have a couple other PRs queued up around the area of these changes.
Thanks.
